jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -1155,4 +1167,27 @@ public String getCharacterEncoding()
             return this.characterEncoding;
         }
     }
+
+    /**
+     * <p>The different modes a welcome resource may be served by a Servlet.</p>
+     */
+    private enum WelcomeServletMode
+    {
+        /**
+         * <p>Welcome resources are not served by Servlets.</p>
+         * <p>The welcome resource must exist as a file on the filesystem.</p>
+         */
+        NONE,
+        /**
+         * <p>Welcome resources may be served by Servlets, but only if the
+         * welcome resource does not exist as a file on the filesystem.</p>
+         */
+        MATCH,
+        /**
+         * <p>Welcome resources may be served by Servlets, but only if the
```

Comment:
I don't think that's correct. If it's `EXACT` we don't check for the existence of the file on disk, only that there is an exact servlet mapping. The text as it is implies that we will check for the file and it must not exist.

Reply:
@janbartel the code in ee9's `DefaultServlet.getWelcomeFile()` and in ee10's `ServletResourceService.getWelcomeTarget()` does check for file existence, so the javadoc reports the current code behavior.
Perhaps the code is wrong, then?